src/twisted/python/runtime.py

```diff
@@ -108,7 +108,7 @@ def isVista(self) -> bool:
         @return: C{True} if the current platform has been detected as Vista
         """
         if getattr(sys, "getwindowsversion", None) is not None:
```

Comment:
```
windowsversion = getattr(sys, "getwindowsversion", None)
return windowsversion is not None and windowsversion[0] == 6
```

Reply:
Is it worth doing this `isinstance` check instead of just ignoring it ?

Reply:
That suggestion would work around the `attr-defined` warning but not the `no-any-return` warning, right?

Reply:
It's certainly odd to lookup the attribute twice

Reply:
```
windowsversion = getattr(sys, "getwindowsversion", None)
return bool(isinstance(windowsversion, tuple) and windowsversion[0] == 6)
```

Reply:
I'd say inefficient, but not odd, since it happens a lot in code.

Actually, that suggestion is incorrect, since `getwindowsversion` is a function, not a tuple.

Reply:
@mthuurne ah of course, I think this should typecheck:

`return sys.platform == "win32" and sys.getwindowsversion().major == 6`

Reply:
yep: https://mypy-play.net/?mypy=latest&python=3.6&flags=strict&gist=bca7bed5e8445b6e8e9e10a1f2aec229